addons/resource/models/resource_calendar.py

```diff
@@ -522,7 +527,8 @@ def _get_attendance_intervals_days_data(self, attendance_intervals):
             interval_hours = (stop - start).total_seconds() / 3600
             day_hours[start.date()] += interval_hours
             if len(self) == 1 and self.flexible_hours:
-                day_days[start.date()] += interval_hours / self.hours_per_day if self.hours_per_day else 0
+                day_hours[start.date()] = meta.duration_hours
+                day_days[start.date()] += meta.duration_hours / self.hours_per_day if self.hours_per_day else 0
```

Comment:
Why not use duration days here?
```suggestion
                day_days[start.date()] += meta.duration_days
```